api/tests/Repositories/TestRepositoryTest.php

```diff
@@ -54,6 +57,24 @@ public function testAll()
         $this->assertGreaterThanOrEqual(5, $result->count());
     }
 
+    public function testAllWithLimit()
+    {
+        $entities = factory(App\Models\TestEntity::class, 5)->create();
+
+        $result = $this->repository->all(['*'], null, 2);
+        $this->assertTrue(is_array($result->toArray()));
+        $this->assertGreaterThanOrEqual(2, $result->count());
```

Comment:
This assertion must be an `assertCount` as if there is more than two the pagination has failed.